Fix Amazon icon not showing up

MariagraziaAlastra opened the pull request:
<!-- Please add the WIP label if the PR isn't complete. -->

**Reviewer:** @andrey-p 

<!-- Optional fields
**CC:**
**Depends on:** 
-->

## Description:
Amazon now shows up as Amazon.com, which breaks the icon.
I made a new shared template to be used wherever we need to normalize company names in order to display the icons. 
Removing the tld using a regex, besides just lowercasing the name, fixed the issue.
<img width="291" alt="screen shot 2018-04-10 at 1 49 55 pm" src="https://user-images.githubusercontent.com/3652195/38577634-8124c332-3ccf-11e8-86be-a4f02f323211.png">
<img width="305" alt="screen shot 2018-04-10 at 1 49 45 pm" src="https://user-images.githubusercontent.com/3652195/38577635-8143b706-3ccf-11e8-8326-c6cf775ad873.png">


## Steps to test this PR:
1. (Optional) force amazon to show up in the top blocked networks
2. Build and reload extension
3. In case 1. was skipped, make sure to load 30 pages. Some with amazon trackers are huffingtonpost.com, washingtonpost.com
4. Make sure the amazon icon shows up in the top blocked pills (if amazon is in the top 3)
5. Also check the tracker networks subview. The icon should show up there as well
6. Everything else should be unaffected


## Automated tests:
- [x] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation


###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

andrey-p commented on shared/js/ui/templates/site.es6.js:
```diff
@@ -1,7 +1,7 @@
 const toggleButton = require('./shared/toggle-button.es6.js')
 const ratingHero = require('./shared/rating-hero.es6.js')
-const trackerNetworksIcon = require('./shared/tracker-network-icon.es6.js')
+const trackerNetworksHeroIcon = require('./shared/tracker-network-hero-icon.es6.js')
```
What's the background of this change? I'm not sure adding `hero` to the name is better. It's used on the main popup, after all.

andrey-p commented on shared/js/ui/templates/shared/normalize-company-icon-name.es6.js:
```diff
@@ -0,0 +1,8 @@
+  companyName = companyName || ''
+  const normalizedName = companyName.toLowerCase().replace(/\.[a-z]+$/, '')
+  return `${normalizedName}`
+}
```
I'm worried we've been pushing logic into the templates. How about putting this in the top blocked model, where we build the list of companies in the first place?

I also think this needs to be tested. We can now add tests to UI models - hit me up if you want more info!

MariagraziaAlastra commented on unit-test/ui/models/mixins/normalize-company-name.es6.js:
```diff
@@ -0,0 +1,39 @@
+let normalizeCompanyName = require('../../../../shared/js/ui/models/mixins/normalize-company-name.es6')
+normalizeCompanyName = normalizeCompanyName.normalizeCompanyName
```
I wanted to do `require('...etc/etc/etc')(normalizeCompanyName)` but then it wouldn't recognize normalizeCompanyName() as a function, so I had to do this on 2 lines.

MariagraziaAlastra commented:
@andrey-p As discussed, I moved the logic to a mixin to be used by both the top-blocked and site-company-list models. Also added unit tests for it!
(Yep I did remember to uncomment the UI stuff in Gruntfile 😄)

andrey-p commented:
This is solid, thank you @MariagraziaAlastra!

Tests :bowing_man:
